rero_ils/modules/documents/tasks.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2021 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Celery tasks to documents."""
+
+from __future__ import absolute_import, print_function
+
+import click
+from celery import shared_task
+# from celery.task.control import inspect
+from flask import current_app
+
+from .api import Document, DocumentsSearch
+from ..contributions.api import Contribution
+
+
+@shared_task(ignore_result=True)
+def find_contribution(verbose=False):
+    """Records creation and indexing.
+
+    :param verbose: Verbose print.
+    :returns: cont_found cunt, cont_exists count,
+              IdRef not cont_found count, MEF not cont_found count
+    """
+    query = DocumentsSearch() \
+        .filter('exists', field='contribution.agent.identifiedBy') \
+        .source(['pid']) \
+        .scan()
+    cont_found = {}
+    cont_exists = {}
+    cont_no_idref = {}
+    cont_no_mef = {}
+    mef_url = current_app.config.get('RERO_ILS_MEF_AGENTS_URL')
+    for hit in query:
+        doc = Document.get_record_by_pid(hit.pid)
```

Comment:
retrieving the `id` tag from ES you can skip one DB call using `get_document_by_id` (instead of `by_pid`)
```
doc = Document.get_record_by_id(hit.meta.id)
```

Reply:
Ok.